src/quest_manager/models.py

```diff
@@ -975,3 +989,142 @@ def get_minutes_to_complete(self):
+            "question to complete the quest."
+        ),
+    )
+    datetime_created = models.DateTimeField(auto_now_add=True)
+    datetime_last_edit = models.DateTimeField(auto_now=True)
+    ordinal = models.PositiveIntegerField(
+        default=1,
+        help_text=(
+            "Indicates the order of the question in the quest so that "
+            "questions are displayed to the student in the correct order."
+        ),
+    )
+
+    quest = models.ForeignKey(Quest, on_delete=models.CASCADE)
+    type = models.CharField(
+        max_length=20, choices=QuestionType.choices, default=QuestionType.SHORT_ANSWER
+    )
+
+    class Meta:
+        ordering = ["ordinal"]
+        # ensure two questions in the same quest cannot have the same ordinal
+        unique_together = ["quest", "ordinal"]
+
+    def __str__(self):
+        """Return string representation in format of: [Quest Name] ([Question Type] - [Question Ordinal])
+        e.g. "Quest 1 (short_answer - 1)" for the first question in the quest with name "Quest 1"
+        """
+        return f"{self.quest.name} ({self.type} - {self.ordinal})"
+
+
+class QuestionSubmission(models.Model):
+    """Used for storing a student's submission for a question.
+
+    This model is used for all types of questions, including multiple choice, short answer, file upload, etc.
+    When a student submits a quest, each Question will have a QuestionSubmission created for it.
+    """
+
+    response_text = models.TextField(
+        null=True,
+        blank=True,
+        help_text=(
+            "A text response to a question. This is what the teacher will "
+            "look at when marking the question."
+        ),
+    )
+    response_file = RestrictedFileField(
+        blank=True,
+        null=True,
+        upload_to="quest/question/submission/%Y/%m/%d",
+        help_text=(
+            "A file response to a question. This is what the teacher will "
+            "look at when marking the question."
+        ),
+    )
+
+    # Similar to a quest's submission being deleted, a question's submission should not be deleted if the question is
+    question = models.ForeignKey(Question, on_delete=models.SET_NULL, null=True)
+
+    # If a quest submission is deleted, the question submissions should also be deleted
+    # This is because semantically they are the same thing in the context of a "Quest Submission"
+    submission = models.ForeignKey(QuestSubmission, on_delete=models.CASCADE)
+
+    class Meta:
+        ordering = ["question__ordinal"]
+
+    def __str__(self):
+        """Return string representation in format of: [Quest Name] ([Question Type] - [Question Ordinal] - [Quest Submission Ordinal])
+        e.g. "Question 1 (short_answer - 1 - 1)" for the first question in the student's first quest submission
+        """
+        if self.question:
+            return f"{self.question.quest.name} ({self.question.type} - {self.question.ordinal} - {self.submission.ordinal})"
```

Comment:
This could just use self.submission (which would cover the quest.name and the submission ordinal) and require less chaining.

Reply:
```py
def __str__(self):
    """Return string representation in format of: [Quest Submission __str__] - ()
    e.g. "Difficult Quest (1) - (1)" for the first question in the student's first quest submission
    """
    if self.question:
        return f"{self.submission} - ({self.question.ordinal})"
    else:
        return f"{self.submission} - (?)"
```